### DISCUSSION TITLE
Suggestion: An example for AutoScaledPool

### DISCUSSION THREAD

**Discussion by EngineerKhan** [apify/crawlee-python#522] [Q&A] [answered]
It would be helpful to have an example for the [AutoScaledPool](https://crawlee.dev/python/api/class/AutoscaledPool#resume). One for Node.JS is [too basic](https://crawlee.dev/api/core/class/AutoscaledPool) and even when trying to replicate it on Python, I am getting the error:

```
TypeError: AutoscaledPool.__init__() missing 4 required keyword-only arguments: 'system_status', 'run_task_function', 'is_task_ready_function', and 'is_finished_function'
```

I am probably missing something, but these arguments aren't provided in the Node.JS example. Can't we use it in a way that only specifies attributes like `concurrency_settings`, `autoscale_interval`, etc? I know playing with these settings is usually not recommended, but I am curious to try it experimentally.

**Comment by vdusek** [comment, marked as answer]
The `AutoscaledPool` is an internal class not intended for end users' direct usage and configuration. However, if you are interested in these internals, you can check out the [BasicCrawler](https://github.com/apify/crawlee-python/blob/master/src/crawlee/basic_crawler/_basic_crawler.py), where `AutoscaledPool` is initialized or check out the test file [test_autoscaled_pool.py](https://github.com/apify/crawlee-python/blob/master/tests/unit/_autoscaling/test_autoscaled_pool.py).

**Reply by EngineerKhan** [reply]
Thank you for both.
